src/components/tl/mlx5/alltoall/alltoall_coll.c

```diff
@@ -242,7 +282,13 @@ static ucc_status_t ucc_tl_mlx5_fanout_start(ucc_coll_task_t *coll_task)
 
     tl_debug(UCC_TASK_LIB(task), "fanout start");
     /* start task if completion event received */
-    UCC_TL_MLX5_PROFILE_REQUEST_EVENT(task, "mlx5_alltoall_fanout_start", 0);
+    if (team->a2a->node.sbgp->group_rank == team->a2a->node.asr_rank) {
```

Comment:
can we remove this if/else when PROFILING is not requested? this is to improve the performance when profiling is not needed